docs/advanced_tutorials/projectors.md

```diff
@@ -0,0 +1,85 @@
+This section introduces the `ProjectorBlock` as an implementation for the the quantum mechanical projection operation onto the subspace spanned by $|a\rangle$: $\mathbb{\hat{P}}=|a\rangle \langle a|$. It evaluates the outer product for bras and kets expressed as bitstrings for a given qubit support. They have to possess matching lengths.
+
+```python exec="on" source="material-block" session="projector" result="json"
+from qadence.blocks import block_to_tensor
+from qadence.operations import Projector  # Projector as an operation.
+
+# Define a projector for |1> onto the qubit labelled 0.
+projector_block = Projector(ket="1", bra="1", qubit_support=0)
+
+# As any block, the matrix representation can be retrieved.
+projector_matrix = block_to_tensor(projector_block)
+
+print(f"projector matrix = {projector_matrix}") # markdown-exec: hide
+```
+
+Other standard operations are expressed as projectors in Qadence. For instance, the number operator $N=\dfrac{1}{2}(I-Z)=|1\rangle\langle 1|$ is used to build $\textrm{CNOT}(i,j)=N(i)\otimes(X(j)-I(j))$. In fact, `CNOT` can also be defined as a projector controlled-unitary operation: $\textrm{CNOT}(i,j)=|0\rangle\langle 0|_i\otimes \mathbb{I}(j)+|1\rangle\langle 1|_i\otimes X(j)$ and we can check their matrix representations are identical:
```

Comment:
I think this equality is not correct: $\textrm{CNOT}(i,j)=N(i)\otimes(X(j)-I(j))$.

```suggestion
Other standard operations are expressed as projectors in Qadence. For instance, the number operator is the projector onto the 1-subspace, $N=|1\rangle\langle 1|$.

In fact, projectors can be used to compose arbitrary operators. For example, the `CNOT` can also be defined as $\textrm{CNOT}(i,j)=|0\rangle\langle 0|_i\otimes \mathbb{I}(j)+|1\rangle\langle 1|_i\otimes X(j)$ and we can check their matrix representations are identical:
```